Composer/packages/client/src/recoilModel/selectors/project.ts

```diff
@@ -115,3 +163,57 @@ export const skillsProjectIdSelector = selector({
     });
   },
 });
+
+export const botProjectDiagnosticsSelector = selector({
+  key: 'botProjectDiagnosticsSelector',
+  get: ({ get }) => {
+    const botProjects = get(botProjectIdsState);
+    const result = botProjects.map((projectId: string) => {
+      const dialogs = get(dialogsSelectorFamily(projectId));
+      const formDialogSchemas = get(formDialogSchemasSelectorFamily(projectId));
+      const luFiles = get(luFilesState(projectId));
+      const lgFiles = get(lgFilesState(projectId));
+      const setting = get(settingsState(projectId));
+      const skillManifests = get(skillManifestsState(projectId));
+      const dialogSchemas = get(dialogSchemasState(projectId));
+      const qnaFiles = get(qnaFilesState(projectId));
+      const botProjectFile = get(botProjectFileState(projectId));
+      const jsonSchemaFiles = get(jsonSchemaFilesState(projectId));
+      const localeSetting = settingStorage.get(projectId);
+      const botAssets: BotAssets = {
+        projectId,
+        dialogs,
+        luFiles,
+        qnaFiles,
+        lgFiles,
+        skillManifests,
+        setting,
+        dialogSchemas,
+        formDialogSchemas,
+        botProjectFile,
+        jsonSchemaFiles,
+        recognizers: [],
+        crossTrainConfig: {},
+      };
+      return BotIndexer.validate(botAssets, localeSetting);
+    });
+    return result;
+  },
+});
+
+export const projectDialogsMapSelector = selector<{ [key: string]: DialogInfo[] }>({
+  key: 'projectDialogsMap',
+  get: ({ get }) => {
+    const projectIds = get(botProjectIdsState);
+    const projectDialogsMap = {};
+
+    projectIds.map((projectId) => {
+      const dialogIds = get(dialogIdsState(projectId));
+      projectDialogsMap[projectId] = dialogIds.map((dialogId) => {
```

Comment:
If you're going to iterate like this over an array, use `forEach`, not `map`. We don't care about the array that's getting returned (and then dropped on the floor) here.